Source/Utils/Source/Event/NativeEventDispatcher.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#include "NativeEventDispatcher.h"
+
+#include <Utils/Logging/Log.h>
+
+#include <memory>
+#include <string>
+
+// -----------------------------------------------------------------------------
+
+DEFINE_LOG_CATEGORY( NativeEventDispatcher, Info )
+
+// -----------------------------------------------------------------------------
+
+namespace Utils
+{
+namespace Internal
+{
+
+// -----------------------------------------------------------------------------
+
+NativeEventDispatcher::NativeEventDispatcher()
+	: EventDispatcher()
+	, EventHandleToContextMap()
+	, EventIdToHandleMap()
+{
+}
+
+// -----------------------------------------------------------------------------
+
+void NativeEventDispatcher::FireEvent( const std::string& EventId, const Event& EventData )
+{
+	auto EventIdToHandleIterator = EventIdToHandleMap.find( EventId );
```

Comment:
Do you think `EventIdToHandleMap` (and `EventIdToHandleIterator`) should be plural?  Maybe `EventIdToHandlesMap`, or preferably if it's not too verbose, `EventIdToListenerHandlesMap`?

I had to take a moment to try to figure out what the purposes of the two maps were for -- I think conceptually it's a bit easier for me to think of `EventIdToHandleMap` as a mapping of `EventId` (which indirectly represents a TEvent) to a list of _listeners_ (where each `EventHandle` represents a unique listener).